src/btree/bt_cursor.c

```diff
@@ -481,19 +482,32 @@ __wt_btcur_search_uncommitted(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp)
      * itself. But, we cannot be sure of finding one, as pre processing of this prepared transaction
      * updates could have happened as part of resolving earlier transaction operations.
      */
-    if (cbt->compare != 0)
-        return (0);
+    if (cbt->compare == 0) {
+        /*
+         * Get the uncommitted update from the cursor. For column store there will be always a
+         * insert structure for updates irrespective of fixed length or variable length.
+         */
+        if (cbt->ins != NULL)
+            upd = cbt->ins->upd;
+        else if (cbt->btree->type == BTREE_ROW)
+            if (cbt->ref->page->modify)
+                upd = cbt->ref->page->modify->mod_row_update[cbt->slot];
+    }
 
     /*
-     * Get the uncommitted update from the cursor. For column store there will be always a insert
-     * structure for updates irrespective of fixed length or variable length.
+     * In the case of prepare, there could be "uncommitted" updates in the lookaside that are newer
+     * than those in the update list.
      */
-    if (cbt->ins != NULL)
-        upd = cbt->ins->upd;
-    else if (cbt->btree->type == BTREE_ROW) {
-        WT_ASSERT(session, cbt->btree->type == BTREE_ROW && cbt->ref->page->modify != NULL &&
-            cbt->ref->page->modify->mod_row_update != NULL);
-        upd = cbt->ref->page->modify->mod_row_update[cbt->slot];
+    if (__wt_page_las_active(session, cbt->ref) && cbt->ref->page_las->has_las &&
+      cbt->ref->page_las->has_prepares) {
```

Comment:
I will say move `has_las` check inside `__wt_page_las_active()`, I think it used to be there at some point. Also, I think you need to explicitly perform `has_prepares` here to be sure that 1. there is las 2. there are prepared

Reply:
@kommiharibabu That function is not enough since it can pass for reasons that have nothing to do with prepare. I've condensed the check down to:
```
cbt->ref->page_las != NULL && cbt->ref->page_las->has_prepares
```
@sulabhM That variable was never part of the check. I added it to replace `las_pageid` since we were comparing it to 0 as a proxy for checking the existence of lookaside entries.